app/assets/javascripts/discourse/app/components/modal/discard-draft.js

```diff
@@ -0,0 +1,25 @@
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
+
+export default class DiscardDraftModal extends Component {
+  @service modal;
+  showSaveDraftButton = this.args.model.allowSaveDraft;
+
+  @action
+  async discardDraft() {
+    await this.args.model.onDestroyDraft();
+    this.args.closeModal();
+  }
+
+  @action
+  async saveDraftAndClose() {
+    await this.args.model.onSaveDraft();
+    this.args.closeModal();
+  }
+
+  @action
+  dismissModal() {
+    this.args.closeModal();
+  }
```

Comment:
IMO it would be best to just reference the `@closeModal` argument directly rather than introducing a wrapper function here.